src/db_window.rs

```diff
@@ -79,7 +81,14 @@ pub fn repair(
         max_entry_height + 2
     };
 
-    let idxs = find_missing_data_indexes(slot, db_ledger, consumed, max_repair_entry_height - 1);
+    let mut idxs = find_missing_data_indexes(
+        slot,
+        db_ledger,
+        consumed,
+        max_repair_entry_height - 1,
+        MAX_REPAIR_LENGTH,
+    );
+    idxs.truncate(MAX_REPAIR_LENGTH);
```

Comment:
why are you truncating if find_missing is limiting its output?

Reply:
oops, I thought I had already removed this, good catch.